crypto/ec/ec_lib.c

```diff
@@ -1265,10 +1266,10 @@ static int ec_field_inverse_mod_ord(const EC_GROUP *group, BIGNUM *r,
     if (!BN_sub(e, group->order, e))
         goto err;
     /*-
-     * Exponent e is public.
-     * No need for scatter-gather or BN_FLG_CONSTTIME.
+     * Although the exponent is public we want the result to be
+     * fixed top.
```

Comment:
Note there is no reason for the two to be correlated like that. Fixed top is about secrecy of the base, while fixed window vs sliding window is about secrecy of the exponent. Tying the two means taking an unnecessary performance hit here.

(`BN_mod_exp_mont_consttime` is fixed window, while the algorithm implemented in the body of `BN_mod_exp_mont` is sliding window. To that end, calling your new function `bn_mod_exp_mont_fixed_top` is a little odd given the existing naming.)

Reply:
Hmm... you're right. On the other hand we are already using the BN_mod_exp_mont_consttime() routine because the input nonce value has BN_FLG_CONSTTIME which makes BN_mod_exp_mont() to call BN_mod_exp_mont_consttime(). I do not think we want to change this in a security fix. We might want to tackle this later.